test/Microsoft.TestPlatform.AcceptanceTests/TranslationLayerTests/EventHandler/DiscoveryEventHandler.cs

```diff
@@ -70,6 +70,10 @@ public class DiscoveryEventHandler2 : ITestDiscoveryEventsHandler2
     /// </summary>
     public List<TestCase> DiscoveredTestCases { get; }
 
+    public IReadOnlyCollection<string> FullyDiscoveredSources { get; private set; }
```

Comment:
Instead of making these properties settable, we could also `Clear` and `AddRange`. This would help to avoid having the list null (if that's what we want).